src/bin/common/solving_loop.ml

```diff
@@ -509,10 +530,57 @@ let main () =
         | None -> print_wrn_opt ~name st_loc "nonnegative integer" value
         | Some i -> Options.set_profiling true i
       end; st
+    | ":optimization", Symbol { name = Simple b; _} ->
+      begin
+        match bool_of_string_opt b with
+        | None -> print_wrn_opt ~name st_loc "bool" value
+        | Some b -> set_optimize b
+      end; st
     | _ ->
       unsupported_opt name; st
   in
 
+  let handle_minimize_term (_term : DStd.Term.t) st =
+    warning "Unsupported minimize.";
+    st
+  in
+  (* TODO: implement when optimae is merged *)
+
+  let handle_maximize_term (_term : DStd.Term.t) st =
+    warning "Unsupported maximize.";
+    st
+  in
+  (* TODO: implement when optimae is merged *)
+
+  let handle_get_objectives (_args : DStd.Term.t list) st =
+    warning "Unsupported get-objectives.";
+    st
+  in
+  (* TODO: implement when optimae is merged *)
+
+  let handle_custom_statement id args st =
+    match id, args with
+    | Dolmen.Std.Id.{name = Simple "minimize"; _}, [term] ->
+      handle_minimize_term term st
+    | Dolmen.Std.Id.{name = Simple "maximize"; _}, [term] ->
+      handle_maximize_term term st
+    | Dolmen.Std.Id.{name = Simple "get-objectives"; _}, args ->
+      handle_get_objectives args st
+    | Dolmen.Std.Id.{name = Simple (("minimize" | "maximize") as ext); _}, _ ->
+      recoverable_error
+        "Statement %s only expects 1 argument (%i given)"
+        ext
+        (List.length args);
+      st
```

Comment:
These should be `when O.get_optimize ()`

Reply:
In fact, instead of using `O.get_optimize ()`, it would be better to:

 - Add a new boolean key `optimize` to the Dolmen state
 - Initialize the boolean key with `O.get_optimize`
 - Use `when State.get st optimize_key` here
 - Update the `optimize_key` on `set-option :optimize` but *not* call `O.set_optimize`
 - Reset the `optimize_key` to `O.get_optimize` when `(reset)` is encountered

This allows the set-option to be local to the Dolmen state, and is correct wrt the semantics of `(reset)` mandated by the SMT-LIB standard (which should reinitialize the state of the solver to what it was at the beginning of parsing the file, so should undo the `set-option` even though it currently doesn't — but we can move towards that slowly).